source/backend/api/Services/CompensationRequisitionService.cs

```diff
@@ -142,7 +151,131 @@ private static string GetCompensationRequisitionStatusText(bool? isDraft)
+            {
+                throw new BadRequestException("Compensation Requisition should have only one parent Id");
+            }
+
+            _user.ThrowInvalidAccessToAcquisitionFile(_userRepository, _acqFileRepository, (long)compensationRequisition.AcquisitionFileId);
+
+            _ = _acqFileRepository.GetById((long)compensationRequisition.AcquisitionFileId) ?? throw new BadRequestException("Invalid acquisitionFileId.");
+
+            compensationRequisition.IsDraft ??= true;
+            var newCompensationRequisition = _compensationRequisitionRepository.Add(compensationRequisition);
+            _compensationRequisitionRepository.CommitTransaction();
+
+            return newCompensationRequisition;
+        }
+
+        private PimsCompensationRequisition AddLeaseFileCompReq(PimsCompensationRequisition compensationRequisition)
+        {
+            _user.ThrowIfNotAuthorized(Permissions.CompensationRequisitionAdd);
+
+            compensationRequisition.ThrowIfNull(nameof(compensationRequisition));
+            if (compensationRequisition.LeaseId is null)
+            {
+                throw new BadRequestException("Invalid LeaseId.");
+            }
+
+            _logger.LogInformation("Adding compensation requisition for lease file id: {leaseId}", compensationRequisition.LeaseId);
+
+            if (compensationRequisition.AcquisitionFileId is not null)
+            {
+                throw new BadRequestException("Compensation Requisition should have only one parent Id");
+            }
+
+            var pimsUser = _userRepository.GetByKeycloakUserId(_user.GetUserKey());
+            var currentLease = _leaseRepository.GetNoTracking((long)compensationRequisition.LeaseId) ?? throw new BadRequestException("Invalid LeaseId.");
+            pimsUser.ThrowInvalidAccessToLeaseFile(currentLease.RegionCode);
+
+            compensationRequisition.IsDraft ??= true;
+            var newCompensationRequisition = _compensationRequisitionRepository.Add(compensationRequisition);
+            _compensationRequisitionRepository.CommitTransaction();
+
+            return newCompensationRequisition;
+        }
+
+        private List<PimsCompensationRequisition> GetAcquisitionFileCompReqs(long fileId)
+        {
+            _logger.LogInformation("Getting compensations for acquisition file id: {acquisitionFileId}", fileId);
+            _user.ThrowIfNotAuthorized(Permissions.CompensationRequisitionView, Permissions.AcquisitionFileView);
+            _user.ThrowInvalidAccessToAcquisitionFile(_userRepository, _acqFileRepository, fileId);
+
+            return _compensationRequisitionRepository.GetAllByAcquisitionFileId(fileId).ToList();
+        }
+
+        private List<PimsCompensationRequisition> GetLeaseFileCompReqs(long fileId)
+        {
+            _logger.LogInformation("Getting compensations for Lease file id: {LeaseId}", fileId);
+            _user.ThrowIfNotAuthorized(Permissions.LeaseView);
```

Comment:
do we need to check for compensation view as well?

Reply:
added.